internal/service/maintenancewindow/resource_maintenance_window.go

```diff
@@ -101,13 +101,10 @@ func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 
 	params := new(admin.GroupMaintenanceWindow)
 
-	if dayOfWeek, ok := d.GetOk("day_of_week"); ok {
-		params.DayOfWeek = cast.ToInt(dayOfWeek)
-	}
+	params.DayOfWeek = cast.ToInt(d.Get("day_of_week"))
```

Comment:
unrelated refactor: day_of_week is required so just simplifying code, same is already done in update operation